MIGRATION.md

```diff
@@ -0,0 +1,126 @@
+## Upgrade to 6.x.x
+
+This new major version of React InstantSearch leverages the [`Context`](https://reactjs.org/docs/context.html) API that has been released part of React 16.3.0. It means that now the **minimal required version for React InstantSearch is React 16.3.0**. We took this opportunity to make React InstantSearch compliant with the [`Strict`](https://reactjs.org/docs/strict-mode.html) mode too. Prior to this version, the lib was not working with those constraints. Here is the list of APIs that have breaking changes:
+
+### SSR
+
+We have made changes to the SSR API to only use the `server` module, well on the server. Previously it was used on both sides: client and server. Those changes in the API also simplifies the flow of information and makes the SSR simpler to reason about. The neat addition that we've made to the API is the `searchClient` hydration. The request was done on the server, but also on the client when the App is mounted. Since the state is passed from the server to the client the initial request on the client was useless. We're now able to avoid this initial request by re-using the results retrieved on the server.
```

Comment:
```suggestion
We made changes to the SSR API to use the `server` module only on the server. It was previously used on both client and server. Those changes in the API also simplify the flow of information and makes the SSR simpler to reason about. The neat addition that we've made to the API is the `searchClient` hydration. The request was done on the server, but also on the client when the app was mounted. Since the state is passed from the server to the client, the initial request on the client was useless. We're now able to avoid this initial request by re-using the results retrieved on the server.
```